packages/core/src/machines/index.ts

```diff
@@ -635,10 +635,10 @@ export const createAuthMachine = ({
           }),
 
         autoSignIn: async () => {
-          if (typeof window !== 'undefined') {
-            const location = window.location
-            if (location.hash) {
-              const params = new URLSearchParams(location.hash.slice(1))
+          if (typeof window !== 'undefined' && window.location) {
+            const hash = window.location.hash
+            if (hash) {
```

Comment:
similar to the early exit where we throw the error, we could make one here too by simply returning when `hash` is `undefined`